src/blocks/textInput/TextInput.tsx

```diff
@@ -0,0 +1,191 @@
+import { ReactNode, forwardRef } from 'react';
+
+import styled, { FlattenSimpleInterpolation } from 'styled-components';
+
+import { ModeProp } from '../Blocks.types';
+
+import { getTextInputState, getTextInputStateStyles } from './TextInput.utils';
+import { useBlocksTheme } from 'blocks/Blocks.hooks';
+import { CrossFilled } from 'blocks/icons';
+import { Text } from 'blocks/text';
+
+export type TextInputProps = {
+  /* Additional prop from styled components to apply custom css to input field */
+  css?: FlattenSimpleInterpolation;
+  /* Render an icon before input field contents */
+  icon?: ReactNode;
+  /* Handles the change in input value */
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  /* Input value */
+  value?: string;
+  /* Input type value */
+  type?: 'text' | 'password';
+  /* Handles the clearing the entire input value */
+  onClear?: () => void;
+  /* Label for the input field */
+  label?: string;
+  /* TotalLength of input value */
+  totalCount?: number;
+  /* Placeholder for input field */
+  placeholder?: string;
+  /* Sets the input field to be compulsory */
+  required?: boolean;
+  /* Sets the input field to error state */
+  error?: boolean;
+  /* Sets the input field to success state */
+  success?: boolean;
+  /* Sets button as disabled */
+  disabled?: boolean;
+  /* Description shown below the input field */
+  description?: string;
+  /* Error message shown below the input field */
+  errorMessage?: string;
+};
+
+const StyledTextInput = styled.div<TextInputProps & ModeProp>`
+  /* Common Input field CSS */
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  gap: var(--s2);
+  .label-count {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+  }
+
+  .input-container {
+    cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
+    display: flex;
+    align-items: center;
+    font-family: var(--font-family);
+    font-size: 14px;
+    font-style: normal;
+    font-weight: 400;
+    line-height: 20px;
+    white-space: nowrap;
+    padding: var(--s0) var(--s3);
+    border-radius: var(--r3);
+
+    /* Common icon css added through CSS class */
+    .icon {
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      margin-right: 2px;
+      span {
+        width: 18px;
+        height: 13.5px;
+      }
+    }
+
+    & input {
+      border: none;
+      background-color: transparent;
+      width: 100%;
```

Comment:
The inout width should be auto or inherit and it should support a width prop to set the custom width. Please check out other design systems on how they work